src/python/py/models/builder.py

```diff
@@ -1018,7 +1018,104 @@ def make_rotary_embedding_caches(self, rotemb, **kwargs):
 
         return cos_cache_name, sin_cache_name
 
+
+    def make_rotary_embedding_partial(self, rotemb, name, root_input, **kwargs):
+        cos_cache_name, sin_cache_name = self.make_rotary_embedding_caches(rotemb)
+
+        #         root_input
+        #             |
+        #          Reshape
+        #             |
+        #           Split
+        #           /   \
+        #       Reshape  |
+        #          |     |
+        #     RoteryEmb  |
+        #          |     |
+        #       Reshape  |
+        #           \   /
+        #           Concat
+        #             |
+        #           Reshape
+
+        num_heads = (self.num_kv_heads if "k_rotary" in name else self.num_attn_heads)
+        dim_size = self.head_size * num_heads
+        const_prefix =  ("k_" if "k_rotary" in name else "v_")
+
+        # Reshape 1
+        reshape_name = f"{name}/Reshape/"
+        reshape_output_0 = f"{name}/Reshape/output_0"
+        rotary_reshape_dim = [0, 0, num_heads, int(dim_size / num_heads)]
+        constant_name = f"{const_prefix}RotaryEmb/Reshape/Const_input_dim_2"
+        constant_value = numpy_helper.from_array(np.array(rotary_reshape_dim, dtype="int64"))
+        self.make_node("Constant", inputs=[], outputs=[f"{constant_name}\constant_0"], name=constant_name, value=constant_value)
```

Comment:
Code blocks like this can be simplified. Constant nodes are [auto-created](https://github.com/microsoft/onnxruntime-genai/blob/main/src/python/py/models/DESIGN.md#constants) when they are detected as a node's input. Nodes such as Reshape can also be created with `make_{op_type}` to simplify adding it to the model.

Here's an example that uses an array for the constant.

https://github.com/microsoft/onnxruntime-genai/blob/0f59a900568cf65d74554cc34027648989f4ab1c/src/python/py/models/builder.py#L1191-L1193

Reply:
I have updated code to use constant via names with data now, this looks really clean. Thanks!